### PR TITLE
Add PR template

### LINKED ISSUE #649: Add PR template

### PR DESCRIPTION
**Pull request by matthew-white**
Closes #649. Similar to the PR templates in central-backend/central/collect.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#### Why is this the best possible solution? Were any other approaches considered?
+
+#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
+
+#### Does this change require updates to user documentation? If so, please file an issue [here](https://github.com/getodk/docs/issues/new) and include the link below.
+
+#### Before submitting this PR, please make sure you have:
+
+- [ ] run `npm run test` and `npm run lint` and confirmed all checks still pass OR confirm CircleCI build passes
+- [ ] verified that any code or assets from external sources are properly credited in comments
```
Comment (matthew-white) [inline .github/PULL_REQUEST_TEMPLATE.md:21]: I've noticed that this checkbox is often left unchecked. Is there a way to reword it to make it clear that in the common case in which there is no external code or assets, it's OK to check the box?

Comment (ktuite) [reply .github/PULL_REQUEST_TEMPLATE.md:21]: 😂 I usually delete this line because it feels wrong to check it, but also not applicable. It would need some kind of definitive "nothing external" clause for me to check it. 

`verified that any code or assets from external sources are properly credited in comments or that everything is internally sourced`

Comment (matthew-white) [reply .github/PULL_REQUEST_TEMPLATE.md]: Added!

**Review by sadiqkhoja — APPROVED** [review]
